source/docs/software/actuators/programming-leds.rst

```diff
@@ -0,0 +1,147 @@
+         :language: java
+         :lines: 8-23
+         :linenos:
+         :lineno-start: 14
+
+   .. group-tab:: C++
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/master/wpilibcExamples/src/main/cpp/examples/AddressableLED/cpp/Robot.cpp
+         :language: cpp
+         :lines: 14-23
+         :linenos:
+         :lineno-start: 14
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/master/wpilibcExamples/src/main/cpp/examples/AddressableLED/cpp/Robot.cpp
+         :language: cpp
+         :lines: 41-47
+         :linenos:
+         :lineno-start: 41
+
+Setting the Entire Strip to One Color
+-------------------------------------
+
+Color can be set to an individual led on the strip using two methods.``setHSV()`` which takes HSV values as an input, and ``setRGB`` which takes RGB values as an input.
+
+Using HSV Values
+^^^^^^^^^^^^^^^^
+
+HSV stands for Hue, Saturation, and Value. Hue describes the color or tint, saturation being the amount of gray, and value being the brightness. Hue is an integer from 1 - 360. Saturation and Value are integers from 0 - 100. LEDs can be set with the ``setHSV`` method that takes 4 arguments: index of the LED, hue, saturation, and value. An example is shown below for setting the color of an LED strip to red.
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: Java
+
+         for (var i = 0; i < m_ledBuffer.getLength(); i++) {
+            // Sets the specified LED to the HSV values for red
+            m_ledBuffer.setHSV(i, 0, 100, 100);
+         }
+
+         m_led.setData(m_ledBuffer);
+
+   .. group-tab:: C++
+
+      .. code-block:: C++
+
+         for (int i = 0; i < kLength; i++) {
+            m_ledBuffer[i].SetHSV(0, 100, 100);
+         }
+
+         m_led.SetData(m_ledBuffer);
+
+Using RGB Values
+^^^^^^^^^^^^^^^^
+
+RGB stands for Red, Green, and Blue. This is a fairly common color model as it's quite easy to understand. LEDs can be set with the ``setRGB`` method that takes 4 arguments: index of the LED, amount of red, amount of green, amount of blue. The amount of Red, Green, and Blue are integer values between 0-255.
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: Java
+
+         for (var i = 0; i < m_ledBuffer.getLength(); i++) {
+            // Sets the specified LED to the RGB values for red
+            m_ledBuffer.setRGB(i, 255, 0, 0);
+         }
+
+         m_led.setData(m_ledBuffer);
+
+   .. group-tab:: C++
+
+      .. code-block:: C++
+
+         for (int i = 0; i < kLength; i++) {
+            m_ledBuffer[i].SetRGB(255, 0, 0);
+         }
+
+         m_led.SetData(m_ledBuffer);
+
+Creating a Rainbow Effect
+-------------------------
+
+It's good robot practice to keep the ``robotPeriodic()`` method as clean as possible, so we'll create a method for handling setting our LED data. We'll call this method ``rainbow()`` and call it from ``robotPeriodic()``.
```

Comment:
I would move this after the next two paragraphs. It's jarring to have a title about creating a rainbow effect, and then the next sentence is about code practices.